doc/glossary.rst

```diff
@@ -0,0 +1,52 @@
+========
+Glossary
+========
+
+Terms used by Nextstrain CLI.  Primarily intended for developers new to the
```

Comment:
_(non-blocking)_

"runner" / "runtime" / "computing platform" is a nice way to describe the different pieces 👌

Defining and standardizing these terms enables us to better communicate how things work both for users and ourselves.

Reply:
:raised_hands: